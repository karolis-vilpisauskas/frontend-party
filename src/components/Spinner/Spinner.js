import React from "react";
import { useSpring, animated } from "react-spring";
import "./Spinner.scss";
const Spinner = () => {
  const props = useSpring({
    to: async (next, cancel) => {
      await next({ opacity: 1 });
      await next({ opacity: 0 });
    },
    from: { opacity: 0 }
  });

  return (
    <animated.div style={props} className="spinner-container">
      <div className="spinner">
        <div></div>
        <div></div>
        <div></div>
        <div></div>
      </div>
    </animated.div>
  );
};

export default Spinner;
