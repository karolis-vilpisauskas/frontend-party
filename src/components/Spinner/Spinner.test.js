import '@testing-library/jest-dom/extend-expect'
import React from "react";
import { cleanup, render } from "@testing-library/react";
import Spinner from "./Spinner";
afterEach(cleanup);

const createTestProps = props => ({
  ...props
});

const renderTest = () => {
  const props = createTestProps();
  const { getByTestId } = render(
    <Spinner {...props}>
    </Spinner>
  );
  const container = getByTestId("spinner");
  return {
    getByTestId,
    container
  };
};

describe("Spinner", () => {
  describe("rendering", () => {
    test("it renders without crashing", () => {
      const { container } = renderTest();
      expect(container).toMatchSnapshot();
    });
  });
});
