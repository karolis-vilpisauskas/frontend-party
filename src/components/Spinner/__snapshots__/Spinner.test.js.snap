// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner rendering it renders without crashing 1`] = `
<div
  class="spinner-container"
  data-testid="spinner"
  style="opacity: 0;"
>
  <div
    class="spinner"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
